lib/src/network/http.rs

```diff
@@ -347,7 +347,10 @@ impl ProxyClient for Client {
     self.metrics().service_start();
 
     if self.back_connected() == BackendConnectionStatus::Connecting {
-      if self.readiness().back_readiness.is_hup() {
+      // We don't yet have the result of the connection to the backend, let's just wait for it..
+      if self.readiness().back_readiness == UnixReady::from(Ready::empty()){
+        return ClientResult::Continue;
```

Comment:
it should not return here: `ready()` might have been called because there were other events on the front socket, and those events must be processed.

Something like this might work better:

```
if self.back_connected() == BackendConnectionStatus::Connecting {
  if self.readiness().back_readiness.is_hup() {
    [...]
  } else if self.readiness().back_readiness != UnixReady::from(Ready::empty()) {
    self.set_back_connected(BackendConnectionStatus::Connected);
  }
}
```

Reply:
Oh yes, I see. I fixed it.